### PR TITLE
QVAC-11763: add  tests for sliding context

### PR DESCRIPTION
**Pull request by donriddo**
## 🎯 What problem does this PR solve?

- Adds comprehensive test coverage for the sliding context feature in LLM addon
- Ensures sliding context behavior is properly validated across different scenarios and edge cases

## 📝 How does it solve it?

- Adds a new integration test file `test/integration/sliding-context.test.js` with 6 test cases covering:
  - Basic sliding context activation when generation exceeds context limit
  - Context overflow error when sliding is disabled (`n_discarded=0`)
  - Multiple sliding activations during long generation sequences
  - Automatic clamping of oversized `n_discarded` values
  - Persistence of `n_discarded` setting across consecutive inference runs
  - Edge case handling with minimal `n_discarded=1` value

## 🧪 How was it tested?

- All 6 new test cases pass, validating sliding context behavior under various configurations
- Tests use a model (Llama-3.2-1B-Instruct-Q4_0) with constrained context size (128) to reliably trigger sliding behavior
- Tests verify both successful sliding activation and proper error handling when sliding is disabled

### CODE REVIEW

File: packages/qvac-lib-infer-llamacpp-llm/test/integration/sliding-context.test.js [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+//
+// Note: llama.cpp may round up ctx_size (e.g. 128 → 256 actual n_ctx).
+// With ~42 prompt tokens, n_predict must exceed available generation space
+// to trigger sliding. n_predict=512 ensures overflow regardless of rounding.
+// ──────────────────────────────────────────────────────────────────────────────
+test('Sliding context allows generation beyond context limit', {
+  timeout: 900_000,
+}, async t => {
+  const { model, logs } = await setupModel(t, {
+    ctx_size: '128',
```
Comment (gianni-cor) [inline packages/qvac-lib-infer-llamacpp-llm/test/integration/sliding-context.test.js]: change to 256 if llama.cpp fallbacks to that one.  you can check in the logs, which one is used by llama-3.2-1b model.  Modify next tests accordingly

File: packages/qvac-lib-infer-llamacpp-llm/test/integration/sliding-context.test.js [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+    entry.includes('discarded') && entry.includes('tokens after the first message')
+  ).length
+}
+
+// ──────────────────────────────────────────────────────────────────────────────
+// Test 1: With n_discarded > 0, generation slides past the context limit
+// instead of throwing a context overflow error.
+//
+// Note: llama.cpp may round up ctx_size (e.g. 128 → 256 actual n_ctx).
+// With ~42 prompt tokens, n_predict must exceed available generation space
```
Comment (gianni-cor) [inline packages/qvac-lib-infer-llamacpp-llm/test/integration/sliding-context.test.js]: keep in mind that n_predict is an upper bound. it could happen that just 10 tokens are generated.  Some tests with a fixed "seed" can help to find if the current prompt is able to reach ctx-size

File: packages/qvac-lib-infer-llamacpp-llm/test/integration/sliding-context.test.js [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+// ──────────────────────────────────────────────────────────────────────────────
+// Test 2: With n_discarded=0 (disabled), generation that exceeds ctx_size
+// triggers a context overflow error from the C++ layer.
+// ──────────────────────────────────────────────────────────────────────────────
+test('Generation fails with context overflow when sliding context is disabled', {
+  timeout: 900_000,
+}, async t => {
+  const { model, logs } = await setupModel(t, {
+    ctx_size: '128',
+    n_predict: '512',
```
Comment (gianni-cor) [inline packages/qvac-lib-infer-llamacpp-llm/test/integration/sliding-context.test.js:146]: always use a seed, to make results deterministic for the same reasons in the above comments

File: packages/qvac-lib-infer-llamacpp-llm/test/integration/sliding-context.test.js [hunk 4]
```diff
@@ -0,0 +1,254 @@
...
+    )
+  }
+
+  t.is(countDiscardLogs(logs), 0, 'no discard events when n_discarded=0')
+})
+
+// ──────────────────────────────────────────────────────────────────────────────
+// Test 3: With a small n_discarded relative to n_predict, sliding activates
+// many times during a single generation. Each activation discards 16 tokens,
+// so with n_predict=1024, sliding must activate at least twice.
```
Comment (gianni-cor) [inline packages/qvac-lib-infer-llamacpp-llm/test/integration/sliding-context.test.js]: given the meaning of n_predict abov, this explanation is not correct. I think we can remove this test

File: packages/qvac-lib-infer-llamacpp-llm/test/integration/sliding-context.test.js [hunk 5]
```diff
@@ -0,0 +1,254 @@
...
+// Test 4: When n_discarded exceeds the available context space (ctx_size -
+// firstMsgTokens), the C++ layer clamps it to (ctx_size - firstMsgTokens - 1)
+// so that at least one token remains after discarding. The inference must
+// succeed without crashing.
+// ──────────────────────────────────────────────────────────────────────────────
+test('Large n_discarded is clamped to fit available context space', {
+  timeout: 900_000,
+}, async t => {
+  const { model, logs } = await setupModel(t, {
+    ctx_size: '128',
```
Comment (gianni-cor) [inline packages/qvac-lib-infer-llamacpp-llm/test/integration/sliding-context.test.js]: use seed

**Review by gianni-cor — CHANGES_REQUESTED** [review]

File: packages/qvac-lib-infer-llamacpp-llm/test/integration/sliding-context.test.js [hunk 6]
```diff
@@ -0,0 +1,254 @@
...
+const FilesystemDL = require('@qvac/dl-filesystem')
+const LlmLlamacpp = require('../../index.js')
+const { ensureModel } = require('./utils')
+const { attachSpecLogger } = require('./spec-logger')
+const os = require('bare-os')
+
+const platform = os.platform()
+const arch = os.arch()
+const isDarwinX64 = platform === 'darwin' && arch === 'x64'
+const isMobile = platform === 'ios' || platform === 'android'
```
Comment (dev-nid) [inline packages/qvac-lib-infer-llamacpp-llm/test/integration/sliding-context.test.js]: `isMobile` is not used anywhere

File: packages/qvac-lib-infer-llamacpp-llm/test/integration/sliding-context.test.js [hunk 7]
```diff
@@ -0,0 +1,254 @@
...
+
+  // Second inference on same instance — n_discarded survives resetState
+  const second = await runAndCollect(model, STORY_PROMPT)
+  t.ok(second.text.length > 0, 'second inference produced output')
+
+  const discardCountAfterSecond = countDiscardLogs(logs)
+  t.ok(
+    discardCountAfterSecond > discardCountAfterFirst,
+    `second run also triggered sliding (total ${discardCountAfterSecond} discards)`
+  )
```
Comment (dev-nid) [inline packages/qvac-lib-infer-llamacpp-llm/test/integration/sliding-context.test.js]: A suggestion, we can make this assertion better by explicitly calculating second-run discards. Then check if they are greater than zero. And if so we can show how many new discards were present in the final message.
Like `second run also triggered sliding (2 new discards, total 5)`

**Comment by donriddo** [comment]
Closed in favour of #85
